txscript/data/script_tests.json

```diff
@@ -1365,87 +1459,193 @@
-["-16 ABS", "-16 NEGATE EQUAL", "NONE", "OK"],
-["0 NOT", "NOP", "NONE", "OK"],
-["1 NOT", "0 EQUAL", "NONE", "OK"],
-["11 NOT", "0 EQUAL", "NONE", "OK"],
-["0 0NOTEQUAL", "0 EQUAL", "NONE", "OK"],
-["1 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
-["111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
-["-111 0NOTEQUAL", "1 EQUAL", "NONE", "OK"],
-["1 1 BOOLAND", "NOP", "NONE", "OK"],
+["BOOLAND test coverage"],
+["NOP BOOLAND", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLAND requires two items on the stack and NOP must not be treated as one"],
+["BOOLAND", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLAND requires two items on the stack, stack has no items"],
+["0 BOOLAND", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLAND requires two items on the stack, stack has one item"],
+["1 1 BOOLAND", "NOP", "NONE", "OK", "BOOLAND must perform a logical and operation on the boolean values of the first two stack items"],
 ["1 0 BOOLAND", "NOT", "NONE", "OK"],
 ["0 1 BOOLAND", "NOT", "NONE", "OK"],
 ["0 0 BOOLAND", "NOT", "NONE", "OK"],
-["16 17 BOOLAND", "NOP", "NONE", "OK"],
-["1 1 BOOLOR", "NOP", "NONE", "OK"],
+["1 2147483648 BOOLAND", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
+
+["BOOLOR test coverage"],
+["NOP BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two input values and NOP must not be treated as one"],
+["BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two input values, stack has no items"],
+["0 BOOLOR", "NOT", "NONE", "ERR_INVALID_STACK_OPERATION", "BOOLOR requires two items on the stack, stack has no items"],
+["1 1 BOOLOR", "NOP", "NONE", "OK", "BOOLOR must perform a logical and operation the boolean values of the first two stack items"],
 ["1 0 BOOLOR", "NOP", "NONE", "OK"],
 ["0 1 BOOLOR", "NOP", "NONE", "OK"],
 ["0 0 BOOLOR", "NOT", "NONE", "OK"],
-["16 17 BOOLOR", "NOP", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUAL", "NONE", "OK"],
-["11 10 1 ADD", "NUMEQUALVERIFY 1", "NONE", "OK"],
-["11 10 1 ADD", "NUMNOTEQUAL NOT", "NONE", "OK"],
+["1 2147483648 BOOLOR", "NOT", "NONE", "ERR_OUT_OF_RANGE", "BOOLAND must fail with stack item >4 bytes"],
+
+["NUMEQUAL test coverage"],
+["NOP", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack and NOP must not be treated as one"],
+["", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack, stack has no items"],
+["-1", "NUMEQUAL", "NONE", "ERR_INVALID_STACK_OPERATION", "NUMEQUAL requires two items on the stack, stack has one item"],
+["11 10 1 ADD", "NUMEQUAL", "NONE", "OK", "NUMEQUAL must compare the first numeric stack item to the second numeric stack item and check if they are equal"],
+["-200 200", "NUMEQUAL NOT", "NONE", "OK"],
+["-1 -1", "NUMEQUAL", "NONE", "OK"],
+["1", "0x02 0x0100 NUMEQUAL", "NONE", "ERR_MINIMAL_DATA", "Numerically equal but rejected by minimal data encoding requirements"],
```

Comment:
These tests for numeric equivalency with rejection due to minimal data encoding should be moved down to the section just below that is specifically dedicated to that.